packages/cache/src/lib/CacheMiddleware.ts

```diff
@@ -0,0 +1,672 @@
+import {
+  ApplyMiddlewareOptions,
+  isEveryByHookPayload,
+  isFilterByHookPayload,
+  isFindByHookPayload,
+  isMapByHookPayload,
+  isPayloadWithData,
+  isRemoveByHookPayload,
+  JoshMiddleware,
+  JoshMiddlewareStore,
+  JoshProvider,
+  Method,
+  Payload,
+  Payloads,
+  PostProvider,
+  PreProvider
+} from '@joshdb/provider';
+import { addExitCallback } from 'catch-exit';
+import { getProperty } from 'property-helpers';
+
+@ApplyMiddlewareOptions({ name: 'cache' })
+export class CacheMiddleware<StoredValue = unknown> extends JoshMiddleware<CacheMiddleware.ContextData<StoredValue>, StoredValue> {
+  public pollingInterval: NodeJS.Timer | undefined;
```

Comment:
```suggestion
  public pollingInterval: NodeJS.Timer | null;
```
Since you are manually setting this to `undefined` later in code, you would rather say, "this is defined, but has no value" with `null` rather than, "this is not defined."

Reply:
It isn't initialized directly in the constructor so undefined would be the correct type

Reply:
```suggestion
  public pollingInterval?: NodeJS.Timer;
```